NYPLCardCreator/Classes/UserSummaryViewController.swift

```diff
@@ -0,0 +1,239 @@
+import UIKit
+
+/// This class is used for summarizing the user's details before
+/// submitting the request to create a library card
+final class UserSummaryViewController: TableViewController {
+  private var cells: [UITableViewCell]
+  private var sectionHeaderTitles: [String]
+  
+  private let configuration: CardCreatorConfiguration
+  private let session: AuthenticatingSession
+  
+  private let homeAddressCell: AddressCell
+  private let altAddressCell: AddressCell
+  private let fullNameCell: UITableViewCell
+  private let emailCell: UITableViewCell
+  private let usernameCell: UITableViewCell
+  private let pinCell: UITableViewCell
+  
+  private let homeAddress: Address
+  private let schoolOrWorkAddress: Address?
+  private let fullName: String
+  private let email: String
+  private let username: String
+  private let pin: String
+  
+  init(
+    configuration: CardCreatorConfiguration,
+    homeAddress: Address,
+    schoolOrWorkAddress: Address?,
+    fullName: String,
+    email: String,
+    username: String,
+    pin: String)
+  {
+    self.configuration = configuration
+    self.session = AuthenticatingSession(configuration: configuration)
+    
+    self.homeAddressCell = AddressCell()
+    self.altAddressCell = AddressCell()
+    self.fullNameCell = UITableViewCell(style: .Default, reuseIdentifier: nil)
+    self.emailCell = UITableViewCell(style: .Default, reuseIdentifier: nil)
+    self.usernameCell = UITableViewCell(style: .Default, reuseIdentifier: nil)
+    self.pinCell = UITableViewCell(style: .Default, reuseIdentifier: nil)
+    
+    self.homeAddress = homeAddress
+    self.schoolOrWorkAddress = schoolOrWorkAddress
+    self.fullName = fullName
+    self.email = email
+    self.username = username
+    self.pin = pin
+    
+    self.homeAddressCell.address = self.homeAddress
+    if let address = self.schoolOrWorkAddress {
+      self.altAddressCell.address = address
+    }
+    self.fullNameCell.textLabel?.text = self.fullName
+    self.emailCell.textLabel?.text = self.email
+    self.usernameCell.textLabel?.text = self.username
+    self.pinCell.textLabel?.text = self.pin
+
+    self.cells = [
+      self.homeAddressCell,
+      self.fullNameCell,
+      self.emailCell,
+      self.usernameCell,
+      self.pinCell
+    ]
+    self.sectionHeaderTitles = [
+      "Home Address",
+      "Full Name",
+      "Email",
+      "Username",
+      "Pin"
+    ]
+    if (self.schoolOrWorkAddress != nil) {
+      self.cells.insert(self.altAddressCell, atIndex: 1)
+      self.sectionHeaderTitles.insert("School or Work Address", atIndex: 1)
+    }
+    
+    super.init(style: .Grouped)
+    self.tableView.estimatedRowHeight = 104
```

Comment:
The summary tableview has a mix of address cells and standard single-line cells. I wasn't able to get correct rendering with a value less than the estimated height for the address cells. If there's a better approach to this let me know.